mono/mini/method-to-ir.c

```diff
@@ -5726,6 +5726,14 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			ins = mono_emit_jit_icall (cfg, mono_get_assembly_object, &assembly_ins);
 			return ins;
 		}
+
+		if (!strcmp (cmethod->name, "GetCallingAssembly")) {
+			// While it is not required per
+			//  https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getcallingassembly(v=vs.110).aspx.
+			// have GetCallingAssembly be consistent independently of varying optimization.
+			// This fixes mono/tests/test-inline-call-stack.cs under FullAOT+LLVM.
+			cfg->no_inline |= COMPILE_LLVM (cfg);
```

Comment:
Could be a one-liner, but the block is prepared to accept fixes for other codegen modes.